examples/demo4.py

```diff
@@ -133,16 +150,32 @@ def main(argv):
         # Take the (e1, e2) shape parameters from the catalog as well.
         psf = psf.shear(e1=cat.getFloat(k,2), e2=cat.getFloat(k,3))
 
-        # Galaxy is a bulge + disk with parameters taken from the catalog:
-        disk = galsim.Exponential(flux=0.6, half_light_radius=cat.getFloat(k,5))
-        disk = disk.shear(e1=cat.getFloat(k,6), e2=cat.getFloat(k,7))
+        # Galaxy is a bulge + disk(+knots) with parameters taken from the catalog:
+
+        # put some fraction of the disk light into knots of star formation
 
-        bulge = galsim.DeVaucouleurs(flux=0.4, half_light_radius=cat.getFloat(k,8))
-        bulge = bulge.shear(e1=cat.getFloat(k,9), e2=cat.getFloat(k,10))
+        disk_hlr = cat.getFloat(k,5)
+        disk_e1 = cat.getFloat(k,6)
+        disk_e2 = cat.getFloat(k,7)
+        bulge_hlr = cat.getFloat(k,8)
+        bulge_e1 = cat.getFloat(k,9)
+        bulge_e2 = cat.getFloat(k,10)
+
+        smooth_disk = galsim.Exponential(flux=smooth_disk_frac, half_light_radius=disk_hlr)
+
+        knots = galsim.RandomWalk(n_knots, disk_hlr, flux=knot_frac)
```

Comment:
Yes, Rachel was right.  The only thing you need is `rng=rng` here (so it's not building a default urandom rng).

However, I think I'd also recommend using somewhat fewer than 100 knots here, since it makes the demo much slower.  (~51 seconds vs ~14 seconds).  Probably the pedagogical impact is achieved with only ~10-20 knots?